src/Lean/Elab/Tactic/Induction.lean

```diff
@@ -174,12 +174,13 @@ private def checkAltNames (alts : Array (Name × MVarId)) (altsSyntax : Array Sy
         throwErrorAt altStx "invalid alternative name '{altName}'"
 
 def evalAlts (elimInfo : ElimInfo) (alts : Array (Name × MVarId)) (optPreTac : Syntax) (altsSyntax : Array Syntax)
+    (initialInfo : Info)
     (numEqs : Nat := 0) (numGeneralized : Nat := 0) (toClear : Array FVarId := #[]) : TacticM Unit := do
```

Comment:
The new test does not exercise the changes at `Induction.lean`. 
I am wondering if we should add a test for them.

Reply:
The tests above for `induction` broke without the change, that's how I found it

Reply:
That is, they broke when I naively changed `getGoals` to `getUnsolvedGoals` in the `induction` code, since the main goal is already assigned at the original location of that code

Reply:
As a reminder, the current logic is to show the initial state everywhere outside `=> rhs` for `induction/cases`. @mhuisi mentioned that it would be nice to show all induction cases after typing e.g. `induction e with`. I don't know if that should mean that we should change the displayed goals for all position outside of `=> rhs`, regardless of whether the syntax tree is complete or not.